built_in_tasks/target_tracking_task.py

```diff
@@ -0,0 +1,447 @@
+'''
+A generic target tracking task
+'''
+import numpy as np
+import time
+import os
+import math
+import traceback
+from collections import OrderedDict
+
+from riglib.experiment import traits, Sequence, FSMTable, StateTransitions
+from riglib.stereo_opengl import ik
+from riglib import plants
+
+from riglib.stereo_opengl.window import Window
+from .target_graphics import *
+
+## Plants
+# List of possible "plants" that a subject could control either during manual or brain control
+cursor = plants.CursorPlant()
+shoulder_anchor = np.array([2., 0., -15])
+chain_kwargs = dict(link_radii=.6, joint_radii=0.6, joint_colors=(181/256., 116/256., 96/256., 1), link_colors=(181/256., 116/256., 96/256., 1))
+chain_20_20_endpt = plants.EndptControlled2LArm(link_lengths=[20, 20], base_loc=shoulder_anchor, **chain_kwargs)
+chain_20_20 = plants.RobotArmGen2D(link_lengths=[20, 20], base_loc=shoulder_anchor, **chain_kwargs)
+
+plantlist = dict(
+    cursor=cursor,
+    chain_20_20=chain_20_20,
+    chain_20_20_endpt=chain_20_20_endpt)
+
+class TargetTracking(Sequence):
+    '''
+    This is a generic cued target tracking skeleton, to form as a common ancestor to the most
+    common type of motor control tracking tasks.
+    '''
+    status = dict(
+        wait = dict(start_trial="target"),
+        target = dict(success="reward", timeout="wait"),
+        reward = dict(reward_end="wait", stoppable=False, end_state=True)
+    )
+
+    # initial state
+    state = "wait"
+    frame_index = 0 #index in the frame in a trial
+    total_distance_error = 0 #Euclidian distance between cursor and target during each trial
+    target_index = -1 # Helper variable to keep track of which target to display within a trial
```

Comment:
i feel like you can simplify a bit by only having 1 target and getting rid of the `target_index` variable. the reason for having two targets in the target capture task was because you might want to have two targets visible at the same time, but i don't think that's ever going to be the case for a tracking task?

Reply:
Yes! You are right, I went back and removed the extra target as it is never used. Thanks!